File: CMS-webAPI/Models/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CMS_webAPI.Models
{
    public class Question
    {
        private string _AuthorId;
        
        public Question()
        {
            this.Quizs = new HashSet<Quiz>();
            this.Comments = new HashSet<QuestionComment>();
            this.Tags = new HashSet<Tag>();
            this.Author = UserService.GetUserViewModelById(this.AuthorId);
        }

        [Key]
        public int QuestionId { get; set; }
        
        [Required]
        public string Description { get; set; }

        [Required]
        public string OptionA { get; set; }

        [Required]
        public string OptionB { get; set; }
                
        public string OptionC { get; set; }
                
        public string OptionD { get; set; }

        [Required]
        public string Answer { get; set; }
        
        public string AnswerDescription { get; set; }
      
        [Required]
        public bool IsLive { get; set; }

        [Required]
        [StringLength(500)]
        public string AuthorId {
            get { return _AuthorId; }
            set { _AuthorId = value;  this.Author = UserService.GetUserViewModelById(value); }
        }

        [Required]
        public DateTime CreatedDate { get; set; }
        [Required]
        public DateTime UpdatedDate { get; set; }

        [Required]
        public int UpdateCount { get; set; }

        [Required]
        public int VisitCount { get; set; }


        //Navigation Properties        
        public virtual ICollection<Quiz> Quizs { get; set; }
        public ICollection<QuestionComment> Comments { get; set; }
        public virtual ICollection<Tag> Tags { get; set; }

        //Not Mapped Properties
        [NotMapped]
        public UserInfoViewModel Author { get; set; }

    }
}

File: CMS-merged/CMS_webAPI/ng_project/app/js/core/domain/Question.js
'use strict';
/*
*	Question
*	Description
*	Question model is for Question entity.
*/

(function() {
	var Question = function(EntityMapper, Tag, User, QuestionComment) {
		function Question(raw) {
			if (raw instanceof Object) {
				this.questionId = raw.QuestionId;				
				this.description = raw.Description;
				this.optionA = raw.OptionA;
				this.optionB = raw.OptionB;
				this.optionC = raw.OptionC;
				this.optionD = raw.OptionD;
				this.answer = raw.Answer;
				this.answerDescription = raw.AnswerDescription;
				this.author = new User(raw.Author);
				this.authorId = raw.AuthorId;
				this.tags = new EntityMapper(Tag).toEntities(raw.Tags);
				this.createdDate = raw.CreatedDate ? new Date(raw.CreatedDate) : undefined;
				this.updatedDate = raw.UpdatedDate ? new Date(raw.UpdatedDate) : undefined;
				this.updateCount = raw.UpdateCount;
				this.visitCount = raw.VisitCount;
				this.isLive = raw.IsLive;
				this.comments = new EntityMapper(QuestionComment).toEntities(raw.Comments);	
			}
		}

		Question.prototype = {			
		};
		return Question;
	};

	Question.$inject = ['EntityMapper', 'Tag', 'User', 'QuestionComment'];
	module.exports = Question;
})();
